packages/styletron-react/src/index.js

```diff
@@ -246,7 +256,10 @@ export function withStyleDeep(component, styleArg) {
     addDebugMetadata(styletron, 2);
   }
 
-  return createStyledElementComponent(autoComposeDeep(styletron, styleArg));
+  return createStyledElementComponent(
+    autoComposeDeep(styletron, styleArg),
+    composeExtension(component, styleArg),
```

Comment:
I have a couple thoughts about this:

1. Should this second argument only be provided in development mode? It seems this extension stuff is only used in development, so I assume this is dead code in production?
2. One reason `createStyledElementComponent` has just a single argument is so that it is composable. In comment below, I suggest adding the extra properties needed for the extension to work on the Styletron object itself. If that is feasible, this could look something like:

```js
return createStyledElementComponent(
  addExtension(autoComposeDeep(styletron, styleArg), component)
);
```

I think this keeps the `createStyledElementComponent` implementation dumber/simpler, since it would only be concerned with single styletron objects, the composition of which is handled by other functions (and allowing for arbitrary composition).

Reply:
Yeah makes perfect sense. I was hoping you might suggest a better way of doing this. I wasn't keen on the way it was being done 👍 I'll make those changes now ⚒